e2e-testing/snakefiles/08_btc_evm_transfer.smk

```diff
@@ -0,0 +1,378 @@
+    input:
+        step_1 = rules.sync_btc_connector.output,
+        btc_connector_file = btc_connector_file,
+        user_account_file = user_account_file,
+        evm_account = evm_account_file,
+        omni_bridge_file = omni_bridge_file
+    output: call_dir / "01_btc_user_deposit_address"
+    params:
+        mkdir = get_mkdir_cmd(call_dir),
+        btc_connector = lambda wc, input: get_json_field(input.btc_connector_file, "contract_id"),
+        user_account_id = lambda wc, input: get_json_field(input.user_account_file, "account_id"),
+        token_locker_id = lambda wc, input: get_json_field(input.omni_bridge_file, "contract_id"),
+        recipient_address = lambda wc, input: get_json_field(input.evm_account, "address"),
+        bridge_sdk_config_file = const.common_bridge_sdk_config_file
+    shell: """
+    {params.mkdir} && \
+         bridge-cli testnet get-bitcoin-address \
+         --chain bitcoin-testnet \
+         --btc-connector {params.btc_connector} \
+         --amount 7300 \
+         -r eth:{params.recipient_address} \
+         --near-token-locker-id {params.token_locker_id} \
+         --near-signer {params.user_account_id} \
+         --config {params.bridge_sdk_config_file} \
+         > {output} \
+    """
+
+rule send_btc_to_deposit_address:
+    message: "Send BTC to user deposit address on Bitcoin"
+    input:
+        step_2 = rules.get_btc_user_deposit_address.output,
+    output: call_dir / "02_send_btc_to_deposit_address"
+    params:
+        scripts_dir = const.common_scripts_dir,
+        btc_address = lambda wc, input: get_btc_address(input.step_2)
+    shell: """
+        node {params.scripts_dir}/send_btc.js {params.btc_address} 7500 > {output}
+    """
+
+rule add_omni_bridge_to_whitelist:
+    message: "Add OmniBridge to whitelist for Post Action in Btc Connector"
+    input:
+        step_1 = rules.sync_btc_connector.output,
+        btc_connector_file = btc_connector_file,
+        near_init_account_file = near_init_account_file,
+        omni_bridge_file = omni_bridge_file
+    output: call_dir / "03_add_omni_bridge_to_whitelist.json"
+    params:
+        mkdir = get_mkdir_cmd(call_dir),
+        scripts_dir = const.common_scripts_dir,
+        btc_connector = lambda wc, input: get_json_field(input.btc_connector_file, "contract_id"),
+        token_locker_id = lambda wc, input: get_json_field(input.omni_bridge_file, "contract_id"),
+        extract_tx = lambda wc, output: extract_tx_hash("near", output)
+    shell: """
+        {params.mkdir} && \
+        {params.scripts_dir}/call-near-contract.sh -c {params.btc_connector} \
+           -m extend_post_action_receiver_id_white_list \
+           -a '{{\"receiver_ids\": [\"{params.token_locker_id}\"]}}' \
+           -f {input.near_init_account_file} \
+           -d "1 yoctoNEAR"\
+           -n testnet 2>&1 | tee {output} && \
+        {params.extract_tx}
+    """
+
+rule add_utxo_chain_connector:
```

Comment:
This and a few other steps also exist in case 04. Do we want to move it to a setup file or have 08 rely on 04?